### PR TITLE
Remove -compat 8.7

### LINKED ISSUE #8383: Removing deprecated notations
Since 8.8, some compatibility aliases are emitting deprecation warnings. The fixes in projects that use them are trivial and clearly explained in the warning message: the typical example is replacing `Nsucc` with `N.succ`.
In order to remove `-compat 8.6` in 8.9, we need projects we test in CI to first fix their code to build with `-w "+compatibility-notation"`.
If that's too much to ask, we could also change the way these deprecation warnings are emitted to make them independent of the compat system (see #8382) and postpone the removal of these aliases to later.

### PR DESCRIPTION
**Pull request by JasonGross**
**Kind:** infrastructure.

Closes #8383

This commit was mostly created by running `./dev/tools/update-compat.py --release`.  There's a bit of manual spacing adjustment around all of the removed compatibility notations, and some test-suite updates were done manually.

- [x] Added / updated test-suite
- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]
My parts `Flags` and `Coqargs` look fine.

**Review by ppedrot — APPROVED** [review]

**Review by jfehrle — APPROVED** [review]
Wording looks OK in release-process.md.

**Comment by Zimmi48** [comment]
The reason for the review request from the doc maintainers was actually trivial: `doc/stdlib/index-list.html.template`.

**Comment by JasonGross** [comment]
What's up with https://gitlab.com/coq/coq/-/jobs/151651683 ?
```
Pulling docker image gitlab/gitlab-runner-helper:x86_64-6e20bd76 ...
ERROR: Job failed: execution took longer than 3h0m0s seconds
```

**Comment by Zimmi48** [comment]
Looks spurious. In any case, CI confirms that we still need coquelicot to be updated (IIRC it only had six lines to fix) and CompCert needs to be updated to use a more recent version of Flocq. Unfortunately, the last released version of Flocq didn't have this patch in, so it's a bit early to recommend CompCert authors they bump their dependency. Looks like in both cases we are waiting on actions by @silene.

**Comment by Zimmi48** [comment]
A new version of Flocq (3.1.0), including the removal of compatibility notations, has been released. It would be nice to know if @xavierleroy has a plan for updating Flocq in the coming months as this would help to plan the integration of this PR.

**Comment by silene** [comment]
For what is worth, I have started working on converting CompCert to Flocq 3.1, I just need some free time to finish the port.

**Comment by Zimmi48** [comment]
That's wonderful!

**Comment by JasonGross** [comment]
Should this PR also add a line to CHANGES about finally removing compatibility notations (possibly with a link to the script to update compat notations... (where is that script, now?))?

**Comment by Zimmi48** [comment]
CHANGES entry: yes
The script is there: https://gist.github.com/JasonGross/9770653967de3679d131c59d42de6d17 (I found the link in CHANGES). You can put it once again, or just refer to the 8.9 CHANGES, given that it needs Coq 8.9 to run.

**Comment by vbgl** [comment]
@silene Any news about CompCert & Flocq 3.1? I can see https://github.com/AbsInt/CompCert/pull/273 but have no clue whether it will be merged.

**Comment by Zimmi48** [comment]
If the PR is not merged soon, we can pin the version we test in CompCert to this specific version. However, there will still be a problem with VST.

**Comment by xavierleroy** [comment]
As a matter of fact, I am looking at this PR https://github.com/AbsInt/CompCert/pull/273 right now.  I still have a few things to discuss with @silene concerning the build procedure.

**Comment by vbgl** [comment]
Thanks for the information!

**Comment by JasonGross** [comment]
Rebased and added a changes entry

**Comment by xavierleroy** [comment]
CompCert trunk was updated to Flocq 3.1, so now you can do what you wanted to do that breaks compatibility with old Flocq versions.

**Comment by vbgl** [comment]
Thank you Xavier!

There is still the coquelicot issue. I’ll submit an overlay unless someone has something ready.

**Comment by vbgl** [comment]
It seems that Coquelicot has been fixed already. What we need is a release or a fix of our CI.

**Comment by Zimmi48** [comment]
We need to unpin our CI. This pinning was done as part of #8322 by @MSoegtropIMC and was related to the Windows packaging.

**Comment by Zimmi48** [comment]
The more complex issue is with VST which must be updated to use the master version of CompCert.
@xavierleroy You don't intend to release a 3.5.1 or 3.6 version of CompCert very soon, do you?

**Comment by MSoegtropIMC** [comment]
@Zimmi48 : Should I make a test run if the master versions of the discussed modules work now or did you do this already? If so, could you describe your findings?

**Comment by Zimmi48** [comment]
I didn't. If you have time for this, please do.

**Comment by MSoegtropIMC** [comment]
> I didn't. If you have time for this, please do.

Ok, but it will have to wait until after the 8.10 feature freeze next week. I hope an addon version update can be done after the freeze, and I want to get in some new things, so this has to be first.

**Comment by Zimmi48** [comment]
This PR cannot be delayed after the freeze, so I just suggest unpinning Coquelicot and commenting out Coquelicot from `dev/ci/gitlab.bat`.

As for VST, I suggest trying out building it with the master version of CompCert like this: https://github.com/PrincetonUniversity/VST/blob/master/BUILD_ORGANIZATION.md#method-b-alternate.

**Comment by Zimmi48** [comment]
>commenting out Coquelicot from dev/ci/gitlab.bat

[This can be fixed after the freeze.]

**Comment by vbgl** [comment]
I’ll give a try to the method B for VST (pun intended). I don’t know what to do for Coquelicot

**Comment by Zimmi48** [comment]
>I don’t know what to do for Coquelicot

In:

**Comment by MSoegtropIMC** [comment]
> I’ll give a try to the method B for VST (pun intended). I don’t know what to do for Coquelicot

Since I am actually using VST in connection with Flocq and CoqInterval I planned to do this cleanup. This is also a follow up on PR #9792. It would be a bit of extra work to do this before #9792 is integrated, because the cleanup requires an independent Flocq, which is added in #9792.

**Comment by vbgl** [comment]
Thank you Michael. PR #9792 is against the `v8.9` branch. Will that change land in the `master` branch too?

**Comment by vbgl** [comment]
So as to avoid delaying the creation of the 8.10 branch, I suggest to keep the compatibility notations that are still in use in VST. We’ll remove them when third-party developers are ready.

File: vernac/g_vernac.mlg
Comment (maximedenes) [inline vernac/g_vernac.mlg]: This should be a catch-all.

**Review by maximedenes — DISMISSED** [review]

**Comment by Zimmi48** [comment]
> So as to avoid delaying the creation of the 8.10 branch, I suggest to keep the compatibility notations that are still in use in VST. We’ll remove them when third-party developers are ready.

This is far from trivial to do because this requires bumping (once again) the version number to which these notations are attached. Another solution would be to open a PR on VST removing the compatibility notations from their internal copy of Flocq. Or we could temporarily change the location of VST that we test to one of our fork. Or we could temporarily disable VST from the CI.

File: vernac/g_vernac.mlg
Comment (Zimmi48) [reply vernac/g_vernac.mlg]: Why? Currently there is a difference between old versions not supported anymore and versions that were never supported. Furthermore, it seems that it is not the right time to request such a change so close to the freeze. This can also be done later on.

Comment (maximedenes) [reply vernac/g_vernac.mlg]: > there is a difference between old versions not supported anymore and versions that were never supported

What's the point of that difference?

> it seems that it is not the right time to request such a change so close to the freeze

Why not?

**Comment by maximedenes** [comment]
> This is far from trivial to do because this requires bumping (once again) the version number to which these notations are attached. Another solution would be to open a PR on VST removing the compatibility notations from their internal copy of Flocq. Or we could temporarily change the location of VST that we test to one of our fork. Or we could temporarily disable VST from the CI.

I don't understand this analysis. Bumping the notations compat number seems much easier than the other solutions. I can do it if necessary.

**Comment by vbgl** [comment]
I’ll try to provide an overlay for VST.

**Comment by MSoegtropIMC** [comment]
A note on VST: it does support (by a build option) to use a pre existing CompCert and (afaik) Flocq. I anyway plan to switch to using this way ob building VST, since it doesn't make that much sense to disconnect VST from numerical verification tools like CoqInterval.

But I have no idea if VST actually compiles with a recent CompCert. But it might be less work to fix this compared to work with an old Flocq (CompCert just updated to use the most recent Flocq).

**Comment by Zimmi48** [comment]
This was the plan initially considered by Vincent. But the freeze date is today, and that's why other options are being considered.

File: vernac/g_vernac.mlg
Comment (JasonGross) [reply vernac/g_vernac.mlg]: These changes are autogenerated by the compatibility script.  Making this change requires both changing the file and updating the script to recognize the new code and to not keep adding things here.  I don't have time to do that today, and I don't think this should delay merging this, and it is important to merge this before branching.  There will be another PR like this submitted shortly after the branch; feel free to make the request there and I'll update things.

Comment (maximedenes) [reply vernac/g_vernac.mlg]: Ok, sounds good, this is not urgent obviously.

**Review by Zimmi48 — APPROVED** [review]
Thanks to everyone! @JasonGross for the initial work, @vbgl for making sure this PR can be ready in time, and all the others. Unfortunately, our CI is a bit overloaded ATM. If I'm not around to merge this when CI is green, @vbgl or anyone else should feel free to re-assign and merge.

**Comment by MSoegtropIMC** [comment]
@vbgl :

> Thank you Michael. PR #9792 is against the `v8.9` branch. Will that change land in the `master` branch too?

A PR for master is pretty top on my todo list, but I can#T promise it this week.

**Comment by vbgl** [comment]
CI failure of ltac2 is because we lost the race against an overlay. The failure of nix I don’t understand; I’ve seen it before so I’m confident it is not specifically related to this PR.

@MSoegtropIMC I’m looking forward your PR. Meanwhile, as VST has been updated, we can merge this without delay.

**Pull request merged**
